automatic/remote-desktop-client/tools/chocolateyInstall.ps1

```diff
@@ -52,8 +52,12 @@ if (-Not (Test-Path -Path $registryPath)) {
 
 Set-ItemProperty -Path $registryPath -Name $name -Value $value -Type DWORD -Force | Out-Null
 
-$uninstallKey    = Get-UninstallRegistryKey -SoftwareName $packageArgs.SoftwareName
-$installLocation = $uninstallKey.InstallLocation
+
+# get install location based on the MSI GUID of the x64 OR x86 version of the app
+# x64 = 37482C7A-E958-455E-938E-0692B5C28708
+# x86 = 74E5B442-57F5-4097-95EA-38E8DBA1EBF1
+
+$installLocation = (Get-WmiObject -Class Win32_Product -Filter 'IdentifyingNumber like "{37482C7A-E958-455E-938E-0692B5C28708}" OR IdentifyingNumber like "{74E5B442-57F5-4097-95EA-38E8DBA1EBF1}"').InstallLocation
```

Comment:
FYI, `Get-WmiObject -Class Win32_Product` is not allowed on the community repository.

https://docs.chocolatey.org/en-us/community-repository/moderation/package-validator/rules/cpmr0059#mainContent

Reply:
Given that `Get-WmiObject` is not permitted another approach is going to be needed here.

The installer is a single installer that contains all of the relevant localization files, and it looks like these are contained in specific country folders beneath the install location - eg. `zh-CN\msrdcw.resources.dll` which contain the localized resource files eg. `RdClient.WPF.Properties.Resources.zh-CN.resources`.

It is actually the `RemoteDesktopTitle` element that is being used to provide the localized element that is mapped to the `DisplayName` entry in the installation registry key.

```
  <data name="RemoteDesktopTitle" xml:space="preserve">
    <value>远程桌面</value>
  </data>
```

I would suggest extracting the `RemoteDesktopTitle` elements from the bundled resources and using these to build a lookup function to map the software name used in the `Get-UninstallRegistryKey` call appropriately.

Reply:
There's maybe another possible way to do this - registry query to HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\